### PR TITLE
firt setup

### PR DESCRIPTION
**Pull request by Juank628**
## To-Do List: first setup
Copied from last week activity...

### CODE REVIEW

File: src/modules/Collection.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  addItem = (description) => {
+    const completed = false;
+    const index = this.items.length + 1;
+    this.items.push({
+      description,
+      completed,
+      index,
+    });
+    this.items.sort((a, b) => a.index - b.index);
+    localStorage.setItem(this.localStorageItem, JSON.stringify(this.items));
```
Comment (paulsaenzsucre) [inline src/modules/Collection.js]: These two lines are repeated in lines 24-25 and lines 45-46. You can kindly refactor in a function to DRY.

File: src/modules/Collection.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  removeItem = (index) => {
+    const arrayIndexToDelete = index - 1;
+    this.items.splice(arrayIndexToDelete, 1);
+    const newItems = this.items.map((item, index) => ({ ...item, index: index + 1 }));
+    this.items = newItems;
+    this.items.sort((a, b) => a.index - b.index);
+    localStorage.setItem(this.localStorageItem, JSON.stringify(this.items));
```
Comment (paulsaenzsucre) [inline src/modules/Collection.js]: You can kindly refactor these lines in a function to DRY.

File: src/modules/Collection.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  getItems = () => this.items.sort((a, b) => a.index - b.index);
+
+  removeAllCompleted = () => {
+    this.items = this.items.filter((item) => item.completed === false);
+    const newItems = this.items.map((item, index) => ({ ...item, index: index + 1 }));
+    this.items = newItems;
+    this.items.sort((a, b) => a.index - b.index);
+    localStorage.setItem(this.localStorageItem, JSON.stringify(this.items));
```
Comment (paulsaenzsucre) [inline src/modules/Collection.js:48]: You can kindly refactor these lines in a function to DRY.

File: src/modules/Render.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    array.forEach((task, i) => {
+      index = i + 1;
+      checked = '';
+      lineThrough = '';
+      if (task.completed) {
+        checked = 'checked';
+        lineThrough = 'style="text-decoration: line-through"';
+      }
+      listItemsHTML += `
+          <li id=list-item-${index} key=${index}><input ${checked} data-index=${index} class="checkbox" type="checkbox"> <input id=description-${index} data-index=${index} ${lineThrough} class="task-description" value="${task.description}" disabled> <img id=more-${index} class="more-icon" data-index=${index} src=${moreIcon} alt=""><img id=delete-${index} class="delete-icon" data-index=${index} src=${deleteIcon} alt=""></li>
```
Comment (paulsaenzsucre) [inline src/modules/Render.js]: This line looks long, you can kindly give it some indenting to better display the HTML code.

**Review by paulsaenzsucre — CHANGES_REQUESTED** [review]
You have done a wonderful job so far! Your code looks clean and well organized. But I found some issues that you can improve. Please, check my inline comments below 👇🏼

File: src/modules/Collection.js (hunk 1 above)
Comment (Juank628) [reply src/modules/Collection.js]: Thanks! Now it's corrected using a function.

File: src/modules/Collection.js (hunk 2 above)
Comment (Juank628) [reply src/modules/Collection.js]: Thanks! Now it's corrected using a function.

File: src/modules/Collection.js (hunk 3 above)
Comment (Juank628) [reply src/modules/Collection.js:48]: Thanks! Now it's corrected using a function.

**Review by paulsaenzsucre — APPROVED** [review]
It's time to merge!!!!

**Comment by Juank628** [comment]
Thank you very much for your comments!!!

**Pull request merged**